app/client/ui/AdminPanel.ts

```diff
@@ -483,6 +481,20 @@ Please log in as an administrator.`),
       ],
     });
   }
+
+  /**
+   * Give an icon summarizing success or failure. Factor in the
+   * severity of the result for failures. This is crude, the
+   * visualization of the results can be elaborated in future.
+   */
+  private _encodeSuccess(result: BootProbeResult) {
+    if (result.success === undefined)  { return '―';  }
+    if (result.success)                { return '✅'; }
+    if (result.severity === 'warning') { return '❗';  }
+    if (result.severity === 'hmm') { return '?';  }
+    // remaining case is a fault.
+    return '❌';
```

Comment:
I wonder if it's worth flipping severity and success around so severity comes first, in case we make a mistake on the backend and return success + warning, for example?

Reply:
I sat down to do this, but I'm not sure if it is right. It could be that the concept of severity is only really relevant if a check fails? To determine whether the failure is an error or a warning or a "hmm" (meaning ~ "debug" I guess). If the check succeeds, is the severity important?

It could be relevant if we were filtering results by severity, so we'd want to see e.g. only things-that-would-be-errors-if-they-failed.

Concretely: if a test with severity `warning` succeeds, what icon should we show?

Reply:
I guess we can choose what `{success: true, severity: "warning"}` means - is it fine or is it a bug, and if fine what should be rendered. I think from your q you'd see it as a bug @Spoffy ?